### PR TITLE
python-telegram-bot: fix vendored urllib patches

### PR DESCRIPTION
**Pull request by pingiun**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Previously using the package would result in an `ImportError`, because one of the vendored urllib3 imports was not replaced.

### CODE REVIEW

File: pkgs/development/python-modules/python-telegram-bot/default.nix [hunk 1]
```diff
@@ -23,7 +23,8 @@ buildPythonPackage rec {
       --replace "import telegram.vendor.ptb_urllib3.urllib3 as urllib3" "import urllib3 as urllib3" \
       --replace "import telegram.vendor.ptb_urllib3.urllib3.contrib.appengine as appengine" "import urllib3.contrib.appengine as appengine" \
       --replace "from telegram.vendor.ptb_urllib3.urllib3.connection import HTTPConnection" "from urllib3.connection import HTTPConnection" \
-      --replace "from telegram.vendor.ptb_urllib3.urllib3.util.timeout import Timeout" "from urllib3.util.timeout import Timeout"
+      --replace "from telegram.vendor.ptb_urllib3.urllib3.util.timeout import Timeout" "from urllib3.util.timeout import Timeout" \
+      --replace "from telegram.vendor.ptb_urllib3.urllib3.fields import RequestField" "from urllib3.fields import RequestField"
```
Comment (jonringer) [inline pkgs/development/python-modules/python-telegram-bot/default.nix]: I feel like this should be upstreamed

Comment (pingiun) [reply pkgs/development/python-modules/python-telegram-bot/default.nix]: An issue at the upstream repository already exists: https://github.com/python-telegram-bot/python-telegram-bot/issues/1583

Comment (jonringer) [reply pkgs/development/python-modules/python-telegram-bot/default.nix]: ... hasn't seen any progress in 4 months... concerning :(

**Comment by jonringer** [comment]
whats the motivation of this? I'm assuming some use case is broken?

**Review by jonringer — APPROVED** [review]
otherwise, the diff LGTM

```
[4 built, 1 copied (0.2 MiB), 0.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/78434
3 package built
python27Packages.python-telegram-bot python37Packages.python-telegram-bot python38Packages.python-telegram-bot
```

Would like to see upstream support in maintaining their package.

**Comment by pingiun** [comment]
> whats the motivation of this? I'm assuming some use case is broken?

The current package is broken, because the import of the vendored package fails. I'm currently looking at the upstream issue to see if I can fix it, but in the mean time this PR should be merged so the nixos package is unbroken

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.python-telegram-bot python37Packages.python-telegram-bot python38Packages.python-telegram-bot

**Pull request merged**